ml-agents/mlagents/trainers/demo_loader.py

```diff
@@ -83,39 +82,40 @@ def demo_to_buffer(
     return brain_params, demo_buffer
 
 
-@timed
-def load_demonstration(
-    file_path: str
-) -> Tuple[BrainParameters, List[AgentInfoActionPairProto], int]:
-    """
-    Loads and parses a demonstration file.
-    :param file_path: Location of demonstration file (.demo).
-    :return: BrainParameter and list of AgentInfoActionPairProto containing demonstration data.
+def get_demo_files(path: str) -> List[str]:
     """
+    Retrieves the demonstration file(s) from a path.
+    :param path: Path of demonstration file or directory.
+    :return: List of demonstration files
 
-    # First 32 bytes of file dedicated to meta-data.
-    INITIAL_POS = 33
-    file_paths = []
-    if os.path.isdir(file_path):
-        all_files = os.listdir(file_path)
-        for _file in all_files:
-            if _file.endswith(".demo"):
-                file_paths.append(os.path.join(file_path, _file))
-        if not all_files:
+    Raises errors if |path| is invalid.
+    """
+    if os.path.isfile(path):
+        if not path.endswith(".demo"):
+            raise ValueError("The path provided is not a '.demo' file.")
+        return [path]
+    elif os.path.isdir(path):
+        fnames = filter(lambda fname: fname.endswith(".demo"), os.listdir(path))
```

Comment:
I generally dislike maps and filters (and the author of python agrees https://www.artima.com/weblogs/viewpost.jsp?thread=98196). I think the old code for this was OK, but if you want to make a change, can you use list comprehensions instead? E.g.
```
fnames = [fname for fname in os.listdir(path) if fname.endswith(".demo")]
```

Reply:
Thanks for the great suggestion! I'd gotten hooked on using `map` and `filter` lately and I had my own "aha" moment that there were clearer list/generator-comprehension alternatives.